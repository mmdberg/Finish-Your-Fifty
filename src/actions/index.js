export const captureUser = (user) => ({
  type: 'CAPTURE_USER',
  user
});

export const addRace = (race) => ({
  type: 'ADD_RACE',
  race
});

export const logOut = () => ({
  type: 'LOG_OUT'
});

export const removeRace = (race) => ({
  type: 'REMOVE_RACE',
  race
});