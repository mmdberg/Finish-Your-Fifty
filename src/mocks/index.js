export const mockRace = {
  venue: 'Atlas Park',
  city: 'San Diego',
  event: 'Spooky 5k',
  date: '04-05-2018'
};

export const mockApiResult = {
  place: {
    placeName: 'Atlas Park',
    cityName: 'San Diego'
  },
  activityStartDate: '2018-04-05T08:00:00',
  organization: {
    organizationName: 'Spooky 5k'
  }
};

export const mockUser = {
  name: 'Pizza',
  email: 'pizza@pizza',
  password: 'p'
};