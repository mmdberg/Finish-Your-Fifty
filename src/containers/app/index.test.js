import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './index';
import { mockUser, mockApiResult, mockRace, mockRaceFromDB, mockCompletedRace } from '../../mocks';
import * as actions from '../../actions';
import { LocalStorage } from '../../__test-helper__/storageMock';

window.localStorage = new LocalStorage();

describe('App', () => {
  let wrapper;
  let mockLogOut;
  let mockCaptureUser;
  let mockAddRace;
  let mockClearRaces;

  beforeEach(() => {
    mockLogOut = jest.fn();
    mockCaptureUser = jest.fn();
    mockAddRace = jest.fn();
    mockClearRaces = jest.fn();
    wrapper = shallow(<App 
      logOut={mockLogOut} 
      captureUser={mockCaptureUser} 
      addRace={mockAddRace}
      clearRaces={mockClearRaces} />,
      { disableLifecycleMethods: true });

  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('local storage', () => {
    beforeEach(() => {
      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        json: () => ([mockRaceFromDB])
      }));
      localStorage.setItem('Last User', JSON.stringify(mockUser));
    });
    
    it('should call capture user if user is in local storage', () => {
      wrapper.instance().componentDidMount()
      expect(mockCaptureUser).toHaveBeenCalledWith(mockUser)
    });

    it('should make fetch with the right params', () => {
      wrapper.instance().componentDidMount();
      expect(window.fetch).toHaveBeenCalledWith('http://localhost:3000/api/v1/races/23');
    });

  //SKIPPED TEST

    it.skip('should call addRace for user\'s races', () => {
      wrapper.instance().componentDidMount();
      expect(mockAddRace).toHaveBeenCalledWith(mockRaceFromDB)
    });
  });

  it('should call logOut on logOut', () => {
    wrapper.instance().logOut();
    expect(mockLogOut).toHaveBeenCalled();
  });

  it('should clear races on logOut', () => {
    wrapper.instance().logOut();
    expect(mockClearRaces).toHaveBeenCalled();
  });

  it('should clear local storage on logOut', () => {
    localStorage.setItem('Last User', JSON.stringify(mockUser));
    wrapper.instance().logOut();
    const lastUser = localStorage.getItem('Last User');
    expect(lastUser).not.toBeDefined();    

  });

  describe('mapStateToProps', () => {
    let mapped; 
    const mockState = {
      user: mockUser
    };

    beforeEach(() => {
      mapped = mapStateToProps(mockState);
    });

    it('correctly maps the user to props', () => {
      expect(mapped.user).toEqual(mockUser);
    });
  });

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let mapped; 

    beforeEach(() => {
      mockDispatch = jest.fn();
      mapped = mapDispatchToProps(mockDispatch);
    });

    it('should call dispatch with correct params on log out', () => {
      const expected = actions.logOut();
      mapped.logOut();
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

    it('should call dispatch with the correct params on captureUser', () => {
      const expected = actions.captureUser(mockUser);
      mapped.captureUser(mockUser);
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

    it('should call dispatch with the correct params on addRace', () => {
      const expected = actions.addRace(mockCompletedRace);
      mapped.addRace(mockCompletedRace);
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

    it('should call dispatch with the correct params on clearRaces', () => {
      const expected = actions.clearRaces();
      mapped.clearRaces();
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

  });

});
