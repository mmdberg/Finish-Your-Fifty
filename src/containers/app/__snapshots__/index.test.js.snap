// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addRace={[Function]}
    captureUser={[Function]}
    clearRaces={[Function]}
    logOut={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Finish Your Fifty
          </h1>
</header>,
        <nav>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-section nav1"
                    exact={true}
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-section nav2"
                    to="/add-race"
          >
                    Add Race
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-section nav3"
                    to="/race-log"
          >
                    Race Log
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-section nav4"
                    to="/search"
          >
                    Search
          </NavLink>
</nav>,
        <div
          className="app-body"
>
          <Switch>
                    <Route
                              exact={true}
                              path="/"
                              render={[Function]}
                    />
                    <Route
                              exact={true}
                              path="/add-race"
                              render={[Function]}
                    />
                    <Route
                              exact={true}
                              path="/race-log"
                              render={[Function]}
                    />
                    <Route
                              exact={true}
                              path="/search"
                              render={[Function]}
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/welcome/login"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/welcome/signup"
                    />
          </Switch>
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="App-title"
>
              Finish Your Fifty
</h1>,
            undefined,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Finish Your Fifty",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Finish Your Fifty",
            "type": "h1",
          },
          undefined,
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-section nav1"
              exact={true}
              to="/"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-section nav2"
              to="/add-race"
>
              Add Race
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-section nav3"
              to="/race-log"
>
              Race Log
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-section nav4"
              to="/search"
>
              Search
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "className": "nav-section nav1",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Add Race",
              "className": "nav-section nav2",
              "to": "/add-race",
            },
            "ref": null,
            "rendered": "Add Race",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Race Log",
              "className": "nav-section nav3",
              "to": "/race-log",
            },
            "ref": null,
            "rendered": "Race Log",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Search",
              "className": "nav-section nav4",
              "to": "/search",
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/add-race"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/race-log"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/search"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/welcome/login"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/welcome/signup"
            />
</Switch>,
          "className": "app-body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/add-race"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/race-log"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/search"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/welcome/login"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/welcome/signup"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/add-race",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/race-log",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/search",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/welcome/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/welcome/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Finish Your Fifty
            </h1>
</header>,
          <nav>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-section nav1"
                        exact={true}
                        to="/"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-section nav2"
                        to="/add-race"
            >
                        Add Race
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-section nav3"
                        to="/race-log"
            >
                        Race Log
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-section nav4"
                        to="/search"
            >
                        Search
            </NavLink>
</nav>,
          <div
            className="app-body"
>
            <Switch>
                        <Route
                                    exact={true}
                                    path="/"
                                    render={[Function]}
                        />
                        <Route
                                    exact={true}
                                    path="/add-race"
                                    render={[Function]}
                        />
                        <Route
                                    exact={true}
                                    path="/race-log"
                                    render={[Function]}
                        />
                        <Route
                                    exact={true}
                                    path="/search"
                                    render={[Function]}
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/welcome/login"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/welcome/signup"
                        />
            </Switch>
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="App-title"
>
                Finish Your Fifty
</h1>,
              undefined,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Finish Your Fifty",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Finish Your Fifty",
              "type": "h1",
            },
            undefined,
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-section nav1"
                exact={true}
                to="/"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-section nav2"
                to="/add-race"
>
                Add Race
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-section nav3"
                to="/race-log"
>
                Race Log
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-section nav4"
                to="/search"
>
                Search
</NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "className": "nav-section nav1",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Add Race",
                "className": "nav-section nav2",
                "to": "/add-race",
              },
              "ref": null,
              "rendered": "Add Race",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Race Log",
                "className": "nav-section nav3",
                "to": "/race-log",
              },
              "ref": null,
              "rendered": "Race Log",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Search",
                "className": "nav-section nav4",
                "to": "/search",
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                            exact={true}
                            path="/"
                            render={[Function]}
              />
              <Route
                            exact={true}
                            path="/add-race"
                            render={[Function]}
              />
              <Route
                            exact={true}
                            path="/race-log"
                            render={[Function]}
              />
              <Route
                            exact={true}
                            path="/search"
                            render={[Function]}
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/welcome/login"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/welcome/signup"
              />
</Switch>,
            "className": "app-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
/>,
                <Route
                  exact={true}
                  path="/add-race"
                  render={[Function]}
/>,
                <Route
                  exact={true}
                  path="/race-log"
                  render={[Function]}
/>,
                <Route
                  exact={true}
                  path="/search"
                  render={[Function]}
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/welcome/login"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/welcome/signup"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/add-race",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/race-log",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/search",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/welcome/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/welcome/signup",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
